### ISSUE TITLE
Duplicate key error on computed property from RegExp literals

### ISSUE THREAD

**Issue by lleaff** [eslint/eslint#8655]
**Tell us about your environment**

* **ESLint Version:** v3.19.0
* **Node Version:** v7.7.2
* **npm Version:** 4.1.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**

<details>

```js
module.exports = {
  "parser": "babel-eslint",

  "globals": {
    "React": true,
    "ReactDOM": true,
    "$": true,
    "_": true,
    "jQuery": true,
  },

  "env": {
    "es6": true,
    "node": true,
    "browser": true,
    "mocha": true,
  },

  "plugins": ["react"],

  "rules": {
    // "indent": [2, 2],
    "linebreak-style": [2, "unix"],
    "semi": [1, "always"], // Enfore semi-colons after statements
    "no-console": [0], // Disallow usage of console API (default 2)
    "no-unused-vars": [1,
      { "vars": "local", // local | all
        "argsIgnorePattern": "^_",
        "args": "after-used", // "after-used": Allow unused params if they come before a used param in the arg list
      }], // Warn against unused variables
    "comma-dangle": [1, "always-multiline"], // Disallow or enforce trailing comma in array literals, e.g.: "always-multiline"
    "no-cond-assign": [1], // Disallow `if (myVar = smth())`

    // eslint-plugin-react
    //------------------------------------------------------------
    "react/display-name": 0, // Prevent missing displayName in a React component definition
    "react/jsx-no-undef": 2, // Disallow undeclared variables in JSX
    "react/jsx-sort-props": 0, // Enforce props alphabetical sorting
    "react/jsx-uses-react": 2, // Prevent React to be incorrectly marked as unused
    "react/jsx-uses-vars": 2, // Prevent variables used in JSX to be incorrectly marked as unused
    "react/no-did-mount-set-state": 2, // Prevent usage of setState in componentDidMount
    "react/no-did-update-set-state": 2, // Prevent usage of setState in componentDidUpdate
    "react/no-multi-comp": 0, // Prevent multiple component definition per file
    "react/no-unknown-property": 2, // Prevent usage of unknown DOM property
    "react/prop-types": 0, // Prevent missing props validation in a React component definition (propTypes)
    "react/react-in-jsx-scope": 2, // Prevent missing React when using JSX
    "react/self-closing-comp": 2, // Prevent extra closing tags for components without children
    //"react/wrap-multilines": 2, // Prevent missing parentheses around multilines JSX // Disabled cause eslint doesn't find the rule (even tho it's documented)

  },

  "extends": ["eslint:recommended"],
};
```
</details>

**What did you do? Please include the actual source code causing the issue.**

```js
const foo = {
  [/bar/]: '/bar/',
  [/foo/]: '/foo/',
};
```

**What did you expect to happen?**
No error as default `toString` implementation of `RegExp` produces different results from different RegExp literals.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  3:4  error    Duplicate key '[object Object]'  no-dupe-keys

✖ 1 problem (1 error, 0 warnings)

```
______

I don't know how practical this would be to implement, but since keys from array literals are correctly differentiated, I'm assuming this shouldn't be hard.

**Comment by soda0289** [comment]
The prop value is cast to a string in the rule `no-dupe-key`
https://github.com/eslint/eslint/blob/735d02d570f4a2d3c663f0f4c6a6d4e4223aeb32/lib/ast-utils.js#L896

I tested the code sample on the eslint demo, which uses the espree parser, and it works as expected. It could be that `babel-eslint` is not generating a RegExp object for the value of Literal nodes.

**Comment by soda0289** [comment]
So this is a bug with `babel-eslint`. When they convert the RegExpLiteral they only set value to an empty Object `{}` and not to a RegExp Object.
https://github.com/babel/babel-eslint/blob/master/babylon-to-espree/toAST.js#L97

I fixed a similar bug in `typescript-eslint-parser` I can work a fix soon.

**Comment by soda0289** [comment]
Fix landed in `babel-eslint`
